addons/account/tests/test_account_payment.py

```diff
@@ -512,3 +513,42 @@ def test_payments_copy_data(self):
 
         self.assertEqual(duplicate_payment_1.amount, payment_1.amount)
         self.assertEqual(duplicate_payment_2.amount, payment_2.amount)
+
+    def test_payments_invoice_payment_state_without_outstanding_accounts(self):
+        """ Ensures that, without an outstanding accounts set on the bank journal payment method,
+            the payment of the invoice still gets a journal entry in community edition"""
+        def register_payment(move, amount, is_community):
+            def patched_installed(self):
+                modules = self.search(['|', ('state', '=', 'installed'), ('name', '=', 'accountant')])
+
+                if is_community:
+                    modules = modules.filtered(lambda m: m.name != 'accountant')
+                return {module.name: module.id for module in modules}
+
+            with patch('odoo.addons.base.models.ir_module.Module._installed', patched_installed):
```

Comment:
```suggestion
            installed = self.env['ir.module.module']._installed
            installed.pop('accountant', None)

            with patch('odoo.addons.base.models.ir_module.Module._installed', installed):
```
But it would probably need to change in favor of `_get_invoice_in_payment_state` anyway